oculus_ros2/src/oculus_sonar_node.cpp

```diff
@@ -1,370 +1,558 @@
+    param_desc.description = "Sonar range (in meters), min=0.3, max=40.0.";
+    param_desc.floating_point_range = {range};
+    this->declare_parameter<double>("range", .3, param_desc);
+  }
+  if (!this->has_parameter("gamma_correction")) {
+    rcl_interfaces::msg::ParameterDescriptor param_desc;
+    rcl_interfaces::msg::IntegerRange range;
+    range.set__from_value(0).set__to_value(255).set__step(1);
+    param_desc.name = "gamma_correction";
+    param_desc.type = rclcpp::ParameterType::PARAMETER_INTEGER;
+    param_desc.description = "Gamma correction, min=0, max=255.";
+    param_desc.integer_range = {range};
+    this->declare_parameter<int>("gamma_correction", 158, param_desc);
+  }
+  if (!this->has_parameter("gain_percent")) {
+    rcl_interfaces::msg::ParameterDescriptor param_desc;
+    rcl_interfaces::msg::FloatingPointRange range;
+    range.set__from_value(0.1).set__to_value(100.0).set__step(0.1);
+    param_desc.name = "gain_percent";
+    param_desc.type = rclcpp::ParameterType::PARAMETER_DOUBLE;
+    param_desc.description = "Gain percentage (%), min=0.1, max=100.0.";
+    param_desc.floating_point_range = {range};
+    this->declare_parameter<double>("gain_percent", 50.0, param_desc);
+  }
+  if (!this->has_parameter("sound_speed")) {
+    rcl_interfaces::msg::ParameterDescriptor param_desc;
+    rcl_interfaces::msg::FloatingPointRange range;
+    range.set__from_value(0.0).set__to_value(1600.0).set__step(0.1);  // min = 1400.0 but must include 0.0 for configuration
+    param_desc.name = "sound_speed";
+    param_desc.type = rclcpp::ParameterType::PARAMETER_DOUBLE;
+    param_desc.description = "Sound speed (in m/s, set to 0 for it to be calculated using salinity), min=1400.0, max=1600.0.";
+    param_desc.floating_point_range = {range};
+    this->declare_parameter<double>("sound_speed", 0.0, param_desc);
+  }
+  if (!this->has_parameter("use_salinity")) {
+    rcl_interfaces::msg::ParameterDescriptor param_desc;
+    param_desc.name = "use_salinity";
+    param_desc.type = rclcpp::ParameterType::PARAMETER_BOOL;
+    param_desc.description = "Use salinity to calculate sound_speed.";
+    this->declare_parameter<bool>("use_salinity", true, param_desc);
+  }
+  if (!this->has_parameter("salinity")) {
+    rcl_interfaces::msg::ParameterDescriptor param_desc;
+    rcl_interfaces::msg::FloatingPointRange range;
+    range.set__from_value(0.0).set__to_value(100.0).set__step(0.1);
+    param_desc.name = "salinity";
+    param_desc.type = rclcpp::ParameterType::PARAMETER_DOUBLE;
+    param_desc.description =
+        "Salinity (in parts per thousand (ppt,ppm,g/kg), "
+        "used to calculate sound speed if needed), min=0.0, max=100";
+    param_desc.floating_point_range = {range};
+    this->declare_parameter<double>("salinity", 0.0, param_desc);
+  }
+
+  // Get the current sonar config
+  update_parameters(currentSonarParameters, this->sonar_driver_->current_ping_config());
```

Comment:
If the function `update_parameters` is used to load download the sonars parameters, it should be called `download_parameters`

Reply:
update_parameters is used to update the local variables currentSonarParameters and currentRosParameters. this->sonar_driver_->current_ping_config() downloads the sonar parameters, this function is only used during initialization.

Reply:
Then rename it `updateLocalParameters`